libs/parallelism/algorithms/include/hpx/parallel/algorithms/reverse.hpp

```diff
@@ -258,3 +360,115 @@ namespace hpx { namespace parallel { inline namespace v1 {
             .call(std::forward<ExPolicy>(policy), first, last, dest_first);
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+    ///////////////////////////////////////////////////////////////////////////
+    // CPO for hpx::reverse
+    HPX_INLINE_CONSTEXPR_VARIABLE struct reverse_t final
+      : hpx::functional::tag_fallback<reverse_t>
+    {
+    private:
+        // clang-format off
+        template <typename BidirIter,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<BidirIter>::value
+            )>
+        // clang-format on
+        friend void tag_fallback_invoke(
+            hpx::reverse_t, BidirIter first, BidirIter last)
+        {
+            static_assert(
+                (hpx::traits::is_bidirectional_iterator<BidirIter>::value),
+                "Requires at least bidirectional iterator.");
+
+            hpx::parallel::v1::detail::reverse<BidirIter>().call2(
+                hpx::execution::sequenced_policy{}, std::true_type{}, first,
+                last);
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename BidirIter,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<BidirIter>::value &&
+                hpx::is_execution_policy<ExPolicy>::value
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy,
+            void>::type
+        tag_fallback_invoke(
+            hpx::reverse_t, ExPolicy&& policy, BidirIter first, BidirIter last)
+        {
+            static_assert(
+                (hpx::traits::is_bidirectional_iterator<BidirIter>::value),
+                "Requires at least bidirectional iterator.");
+
+            typedef hpx::is_sequenced_execution_policy<ExPolicy> is_seq;
+
+            return parallel::util::detail::algorithm_result<ExPolicy>::get(
+                hpx::parallel::v1::detail::reverse<BidirIter>().call2(
+                    std::forward<ExPolicy>(policy), is_seq(), first, last));
```

Comment:
I think you can remove the `is_seq` predicate here and simply rely on the the dispatch facility to figure it out:

```suggestion
            return parallel::util::detail::algorithm_result<ExPolicy>::get(
                hpx::parallel::v1::detail::reverse<BidirIter>().call(
                    std::forward<ExPolicy>(policy), first, last));
```

Reply:
ok I was going to ask you about `call2`. thanks!